src/main/ruby/truffleruby/core/struct.rb

```diff
@@ -56,7 +58,7 @@ def self.new(klass_name, *attrs, keyword_init: nil, &block)
     end
 
     klass = Class.new self do
-      _specialize attrs unless keyword_init
+      _specialize attrs unless use_keyword_init
```

Comment:
Is this what we want? (Run `_specialize` when `nil` or `true`?)

Reply:
`_specialize` doesn't handle keywords (currently) so it should be only if `keyword_init: false`.
`use_keyword_init` seems a bit confusing (kwargs might or not be used for `nil`) so I would just move the test here.

Which is unfortunate because then `Struct.new(:a, :b)` will not use `_specialize` anymore with this PR.
That seems suboptimal for performance, although the main thing `_specialize` does is optimizing `initialize`.
OTOH it seems worth it for compatibility.

I think we should use an approach similar to Data longer term.

Reply:
I had originally planned to use `use_keyword_init` in more places and then eventually only needed it in one but didn't factor it out.